src/util/lifecycle_utils.js

```diff
@@ -69,7 +69,154 @@ function file_matches_filter({obj_info, filter_func = undefined}) {
     return true;
 }
 
+/**
+ * get_lifecycle_rule_for_object determines the most specific matching lifecycle rule for the given object metadata
+ *
+ * priority is based on:
+ *  - longest matching prefix
+ *  - most matching tags
+ *  - narrowest object size range
+ *
+ * @param {Array<Object>} rules
+ * @param {Object} object_info
+ * @returns {Object|undefined}
+ */
+function get_lifecycle_rule_for_object(rules, object_info) {
+    let matched_rule;
+    let rule_priority = {
+        prefix_len: -1,
+        tag_count: -1,
+        size_span: Infinity,
+    };
+    if (object_info?.key && rules?.length > 0) {
+        for (const rule of rules) {
+            if (rule?.status !== 'Enabled') continue;
+
+            const filter = rule?.filter || {};
+
+            const filter_func = build_lifecycle_filter(filter);
+
+            if (!filter_func(object_info)) { continue; }
+
+            const priority = {
```

Comment:
can you move the handling of priority to a function. just for organization

Reply:
Sure, thanks